### PR TITLE
⚠ ClusterExtension: add installNamespace, remove watchNamespaces

### PR DESCRIPTION
**Pull request by joelanford**
# Description

In `ClusterExtension`:
- Add `spec.installNamespace` as a required field. This will be used to configure the context of the client that templates and applies the bundle contents. This will also be necessary for the introduction of a `spec.serviceAccount` field that operator-controller will use to perform all interactions with the apiserver for that ClusterExtension.
- Remove `spec.watchNamespaces` and hardcode all registry+v1 based bundles to watch all namespaces (or fail if they don't support AllNamespaces mode)

## Reviewer Checklist

- [ ] API Go Documentation
- [ ] Tests: Unit Tests (and E2E Tests, if appropriate)
- [ ] Comprehensive Commit Messages
- [ ] Links to related GitHub Issue(s)

### CODE REVIEW

**Comment by joelanford** [comment]
/hold

I want to add a few more tests that are specific to the use of the new InstallNamespace field.

File: internal/controllers/clusterextension_admission_test.go [hunk 1]
```diff
@@ -186,6 +190,52 @@ func TestClusterExtensionAdmissionChannel(t *testing.T) {
...
+	testCases := []struct {
+		name             string
+		installNamespace string
+		errMsg           string
+	}{
+		{"just alphanumeric", "justalphanumberic1", ""},
+		{"hypen-separated", "hyphenated-name", ""},
+		{"no install namespace", "", regexMismatchError},
+		{"dot-separated", "dotted.name", regexMismatchError},
+		{"includes version", "channel-has-version-1.0.1", regexMismatchError},
```
Comment (everettraven) [inline internal/controllers/clusterextension_admission_test.go:206]: We already check for dot separated, I don't think we need to explicitly check for "includes version" for install namespaces

```suggestion
```

**Review by everettraven — APPROVED** [review]
Looks good. One minor nit, but IMO not something that is worth holding this PR on and will leave up to you if you'd like to address.

File: internal/controllers/clusterextension_admission_test.go (hunk 1 above)
Comment (joelanford) [reply internal/controllers/clusterextension_admission_test.go:206]: Yep good point, this was a copy/paste job from the channel name tests, and this one does not seem relevant here. I'll remove.

**Review by everettraven — APPROVED** [review]

**Pull request merged**